awpy/analytics/map_control.py

```diff
@@ -0,0 +1,413 @@
+    )
+
+
+def _calc_map_control_metric_from_dict(
+    map_name: str,
+    mc_values: FrameMapControlValues,
+) -> float:
+    """Return map control metric given FrameMapControlValues object.
+
+    Map Control metric is used to quantify how much of the map is controlled
+    by T/CT. Each tile is given a value between -1 (complete T control) and 1
+    (complete CT control). If a tile is controlled by both teams, a value is
+    found by taking the ratio between the sum of CT values and sum of CT and
+    T values. Once all of the tiles' values are calculated, a weighted sum
+    is done on the tiles' values where the tiles' area is the weights.
+    This weighted sum is transformed to fit the range [-1, 1] and then
+    returned as the map control metric.
+
+    Args:
+        map_name (str): Map used in calculate_tile_area
+        mc_values (FrameMapControlValues): Object containing map control
+            values for both teams.
+            Expected format that of calc_frame_map_control_values output
+
+    Returns: Map Control Metric
+    """
+    current_map_control_value: list[float] = []
+    tile_areas: list[float] = []
+    for tile in set(mc_values.ct_values) | set(mc_values.t_values):
+        ct_val, t_val = mc_values.ct_values[tile], mc_values.t_values[tile]
+
+        current_map_control_value.append(sum(ct_val) / (sum(ct_val) + sum(t_val)))
+        tile_areas.append(calculate_tile_area(map_name, int(tile)))
+
+    np_current_map_control_value = np.array(current_map_control_value)
+    np_tile_areas = np.array(tile_areas)
+
+    return (
+        (sum(np_current_map_control_value * np_tile_areas) / sum(np_tile_areas)) * 2
+    ) - 1
+
+
+def calc_frame_map_control_metric(
+    map_name: str,
+    frame: GameFrame,
+) -> float:
+    """Return map control metric for given awpy frame.
+
+    Map Control metric is used to quantify how much of the map is controlled
+    by T/CT. Each tile is given a value between -1 (complete T control) and 1
+    (complete CT control). If a tile is controlled by both teams, a value is
+    found by taking the ratio between the sum of CT values and sum of CT and
+    T values. Once all of the tiles' values are calculated, a weighted sum
+    is done on the tiles' values where the tiles' area is the weights.
+    This weighted sum is transformed to fit the range [-1, 1] and then
+    returned as the map control metric.
+
+    Args:
+        map_name (str): Map used position_transform call
+        frame (GameFrame): awpy frame to calculate map control metric for
+
+    Returns: Map Control metric for given frame
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    map_control_values = calc_frame_map_control_values(map_name, frame)
+
+    return _calc_map_control_metric_from_dict(map_name, map_control_values)
+
+
+def calculate_round_map_control_metrics(
```

Comment:
bit confused here -- for whom do the map control values returned by this function belong to?

Reply:
This function is returning a list of map control metrics for each frame in a passed in awpy round. It's not used elsewhere in the awpy codebase, but I use it in the notebook to easily calculate the map control metric throughout the entire round. Should I remove it?

Reply:
Ok, yes, I understand it now. I think it is fine, especially since it is showcased in the notebook